### PR TITLE
Fixed code

### PR DESCRIPTION
**Pull request by lyrixx**
and also remove dead code

### CODE REVIEW

File: src/Cloneable.php [hunk 1]
```diff
@@ -11,7 +11,9 @@ public function with(...$values): static
         $clone = (new ReflectionClass(static::class))->newInstanceWithoutConstructor();
 
         foreach ($this as $objectField => $objectValue) {
-            $objectValue = $values[$objectField] ?? $objectValue;
+            if (array_key_exists($objectField, $values)) {
```
Comment (freekmurze) [inline src/Cloneable.php:14]: This refactor doesn't seem to have any benefit. I think the previous code is readable enough.

Comment (lyrixx) [reply src/Cloneable.php:14]: The previous code does not work in all situation. I added a test to prove it.
So this is not a refactor, this is a bug fix 😎

**Comment by lyrixx** [comment]
ping @brendt

**Comment by brendt** [comment]
I've already changed it myself before I saw this PR, thanks anyway! (https://github.com/spatie/php-cloneable/blob/main/src/Cloneable.php#L14)

@freekmurze `??` wouldn't work if you'd pass `null` to `with`, so this is a legitimate change.

**Pull request closed without merging**